### PR TITLE
Privileges test

### PR DESCRIPTION
**Pull request by Sodman**
Add unit test to ensure all deployments are run as non-root by default.

For reference, any runAsUser set at the container level will override a runAsUser set at the pod level. Therefore, any of the following configs is acceptable.

- Have nonRoot user set at the pod level security context
- Also have a nonRoot user set at the container level security context

or

- Have nonRoot user set at the pod level security context
- Have no explicit configuration set at the container level security context

or

- Have no explicit user set at the pod level security context
- Have a nonRoot user explicitly set at container level for _every_ container in the pod

### CODE REVIEW

**Comment by Sodman** [comment]
/kick e2e test flake https://console.cloud.google.com/cloud-build/builds/94592189-616f-4956-b39f-1397722ff5c9;step=14?authuser=1&project=solo-public

>[Fail] Kube2e: gateway tests with virtual service with a mix of valid and invalid virtual services [It] preserves the valid virtual services in envoy when a virtual service has been made invalid 
/go/pkg/mod/github.com/onsi/ginkgo@v1.11.0/internal/leafnodes/runner.go:113

File: install/test/privileges_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	v1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var _ = Describe("Deployment Privileges Test", func() {
+	var allTests = func(testCase renderTestCase) {
+		Context("Gloo", func() {
+			Context("all cluster-scoped deployments", func() {
+				It("is running all deployments with non root user permissions by default", func() {
+					testManifest, err := testCase.renderer.RenderManifest(namespace, helmValues{})
```
Comment (kdorosh) [inline install/test/privileges_test.go:18]: same comment about providing some helm overrides here to test more deployments (e.g., enable ingress, knative, access logger, mtls)

**Comment by kdorosh** [comment]
added WIP label so PR doesn't merge if PR passes CI and @Sodman addresses my comment

File: install/test/privileges_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+						},
+					})
+					Expect(err).NotTo(HaveOccurred(), "Should be able to render the manifest in the priviliges unit test")
+
+					expectNonRoot(testManifest)
+				})
+			})
+		})
+	}
+	runTests(allTests)
```
Comment (kdorosh) [inline install/test/privileges_test.go:42]: there's no need for the `allTests` variable or for wrapping `Context`. The tests would run regardless.

we only wrap them in a variable if we want to run the same set of tests more than once, with different setups (e.g., run each helm test for helm 2 and helm 3)

**Comment by Sodman** [comment]
/kick [This known flake](https://github.com/solo-io/gloo/issues/3226). Logs: https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=70cb66c4-7a63-465f-9c41-fee9b4108248

**Pull request merged**